staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -456,6 +457,62 @@ func isDryRun(url *url.URL) bool {
 	return len(url.Query()["dryRun"]) != 0
 }
 
+type fieldValidationDirective int
+
+const (
+	ignoreFieldValidation fieldValidationDirective = iota
+	strictFieldValidation
+	warnFieldValidation
+)
+
+// fieldValidation checks if the fieldValidation query parameter is set on the request,
+// and if so ensures that the parameter is valid and that the request has a valid
+// media type, because the list of media types that support field validation are a subset of
+// all supported media types (only json and yaml supports field validation).
+func fieldValidation(req *http.Request) (fieldValidationDirective, error) {
+	supportedMediaTypes := []string{"application/json", "application/yaml"}
```

Comment:
Aha, I thought from reading the KEP that we would not support this way, and hence, keep the API area as small as possible to begin with at least?

Reply:
Ultimately, I’m just trying to make sure we error if the user requests validation on protobuf content. 

I figured it would be more correct to have an explicit allowlist of json and yaml rather than a denylist of protobuf. I’m not sure if that’s better or not though, since protobuf is the only disallowed content type (that wouldn’t trigger an unsupported content type error later on). Let me know what you think is the best way to do this